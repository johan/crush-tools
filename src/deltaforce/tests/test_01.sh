test_number=01
description="defaults"

left=$test_dir/test_$test_number-full.txt
right=$test_dir/test_$test_number-delta.txt
expected=$test_dir/test_$test_number.expected

output=$test_dir/test_$test_number.0.out
$bin $left $right > $output

if [ $? -ne 0 ] ||
   [ "`diff -q $output $expected`" ]; then
  test_status $test_number 0 "$description (${subtests[0]})" FAIL
else
  test_status $test_number 0 "$description (${subtests[0]})" PASS
  rm "$output"
fi

output=$test_dir/test_$test_number.1.out
cat $left | $bin - $right > $output

if [ $? -ne 0 ] ||
   [ "`diff -q $output $expected`" ]; then
  test_status $test_number 1 "$description (${subtests[1]})" FAIL
else
  test_status $test_number 1 "$description (${subtests[1]})" PASS
  rm "$output"
fi

output=$test_dir/test_$test_number.2.out
cat $right | $bin $left - > $output

if [ $? -ne 0 ] ||
   [ "`diff -q $output $expected`" ]; then
  test_status $test_number 2 "$description (${subtests[2]})" FAIL
else
  test_status $test_number 2 "$description (${subtests[2]})" PASS
  rm "$output"
fi

