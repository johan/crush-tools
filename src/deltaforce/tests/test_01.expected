ID	Time
1	10-29-2007-11:39:17
2	11-10-2007-19:55:28
3	10-26-2007-15:39:56
4	10-30-2007-16:53:36
5	11-10-2007-12:29:18
6	11-09-2007-12:40:31
7	11-10-2007-12:40:31
